src/DIRAC/WorkloadManagementSystem/Client/JobState/JobState.py

```diff
@@ -441,3 +444,16 @@ def insertIntoTQ(self, manifest=None):
                     gLogger.info("Job %s removed from the TQ" % self.__jid)
             return S_ERROR("Cannot insert in task queue: %s" % errMsg)
         return S_OK()
+
+    def getJobPath(self):
+        """
+        Get JobPath attribute from manifest if founded, otherwise return None
+
+        :return: list[str] / None
+        """
+        result = self.getManifest()
+        if result["OK"]:
```

Comment:
if `result` is an `S_ERROR`, the error should be propagated. This also means that this method should return S_OK() or S_ERROR.